src/Compilers/Core/Portable/CommandLine/CommonCompiler.cs

```diff
@@ -559,304 +571,337 @@ private int RunCore(TextWriter consoleOutput, ErrorLogger errorLogger, Cancellat
+            if (diagnostics.HasAnyErrors())
+            {
+                return;
+            }
 
-                string outputName = GetOutputFileName(compilation, cancellationToken);
-                var finalPeFilePath = Path.Combine(Arguments.OutputDirectory, outputName);
-                var finalPdbFilePath = Arguments.PdbPath ?? Path.ChangeExtension(finalPeFilePath, ".pdb");
-                var finalXmlFilePath = Arguments.DocumentationPath;
+            DiagnosticBag analyzerExceptionDiagnostics = null;
 
-                var diagnosticBag = DiagnosticBag.GetInstance();
-                NoThrowStreamDisposer sourceLinkStreamDisposerOpt = null;
+            if (!analyzers.IsEmpty)
+            {
+                analyzerCts = CancellationTokenSource.CreateLinkedTokenSource(cancellationToken);
+                analyzerManager = new AnalyzerManager(analyzers);
+                analyzerExceptionDiagnostics = new DiagnosticBag();
+                Action<Diagnostic> addExceptionDiagnostic = diagnostic => analyzerExceptionDiagnostics.Add(diagnostic);
+                var analyzerOptions = new AnalyzerOptions(ImmutableArray<AdditionalText>.CastUp(additionalTextFiles));
+
+                analyzerDriver = AnalyzerDriver.CreateAndAttachToCompilation(
+                    compilation,
+                    analyzers,
+                    analyzerOptions,
+                    analyzerManager,
+                    addExceptionDiagnostic,
+                    Arguments.ReportAnalyzer,
+                    out compilation,
+                    analyzerCts.Token);
+                reportAnalyzer = Arguments.ReportAnalyzer && !analyzers.IsEmpty;
+            }
 
-                try
+            compilation.GetDiagnostics(CompilationStage.Declare, includeEarlierStages: false, diagnostics, cancellationToken);
+            if (diagnostics.HasAnyErrors())
+            {
+                return;
+            }
+
+            cancellationToken.ThrowIfCancellationRequested();
+
+            string outputName = GetOutputFileName(compilation, cancellationToken);
+            var finalPeFilePath = Path.Combine(Arguments.OutputDirectory, outputName);
+            var finalPdbFilePath = Arguments.PdbPath ?? Path.ChangeExtension(finalPeFilePath, ".pdb");
+            var finalXmlFilePath = Arguments.DocumentationPath;
+
+            NoThrowStreamDisposer sourceLinkStreamDisposerOpt = null;
+
+            try
+            {
+                // NOTE: Unlike the PDB path, the XML doc path is not embedded in the assembly, so we don't need to pass it to emit.
+                var emitOptions = Arguments.EmitOptions.
+                    WithOutputNameOverride(outputName).
+                    WithPdbFilePath(PathUtilities.NormalizePathPrefix(finalPdbFilePath, Arguments.PathMap));
+
+                // This feature flag is being maintained until our next major release to avoid unnecessary 
```

Comment:
>next major releas [](start = 67, length = 17)

This is not introduced by this PR, but which release is the "next major release". Was it 15.0? Is it 16.0?
Could you file a follow-up issue?

Reply:
I believe that is this issue: https://github.com/dotnet/roslyn/issues/19592